lib/pure/smtp.nim

```diff
@@ -141,31 +153,92 @@ proc close*(smtp: Smtp) =
   smtp.debugSend("QUIT\c\L")
   smtp.sock.close()
 
-proc createMessage*(mSubject, mBody: string, mTo, mCc: seq[string],
+
+proc getMimeTypeOfFile(pathToFile:string): string =
+  ## This will return the mimetype
+  var mimeDB = newMimetypes()
+  var ext = extractExt(pathToFile) # 
+
+  ## we choose "application/octet-stream" 
+  ## if no extension is given
+  ## TODO what about linux? or files without extension.
+  ## Do we have to have something like `file` magic?
+  return mimeDB.getMimetype(ext,"bin") 
+
+
+
+proc composeAttachement(attachement:Attachment,boundary:string) : string=
+  ## this generates an attachement
+  result =""
+
+  var ContentDisposition :string = "Content-Disposition: attachment;"
+  if attachement.filename != nil:
+    ContentDisposition &= " filename=" & attachement.filename & ";"
+
+  result.add("\c\L")
+  result.add("Content-Type: " & attachement.mimeType & "\c\L")
+  result.add( ContentDisposition & "\c\L")
+  result.add("MIME-Version: 1.0\c\L")
+  result.add("Content-Transfer-Encoding: base64 \c\L")
+  result.add("" & "\c\L")
+  result.add(attachement.contentB64 & "\c\L")
+  result.add("\c\L")
+  result.add("--" & boundary )
+
+
+proc attach*(msg:var Message,pathToFile:string,
+              myFilename:string) =
+  ## this will attach a file to the message
+  var attachement = Attachment()
+  attachement.mimeType    = getMimeTypeOfFile(pathToFile)
+  let content             = readFile(pathToFile)
+  attachement.contentB64  = encode(content)
+  attachement.filename    = myFilename
+  #attachement.description = description
+  add(msg.attachements,attachement)
+
+
+proc getRandomBoundary():string = 
+  var mt = newMersenneTwister( int( epochTime() )  ) #we dont need stong random
+  return $mt.getNum()
+
+
+proc createMessage*(mSubject, mBody: string, mFrom:string, 
```

Comment:
Ahh. Could you define the old procedure (without the ``mFrom`` param) but make it deprecated and explain why in the doc comment for that procedure?